src/updatechecker.cpp

```diff
@@ -219,9 +220,34 @@ UpdateChecker::UpdateChecker(): Thread("WinSparkle updates check")
 
 void UpdateChecker::Run()
 {
-    // no initialization to do, so signal readiness immediately
-    SignalReady();
+    while (1)
+    {
+        // no initialization to do, so signal readiness immediately
+        SignalReady();
+
+        bool checkUpdates;
+        if (Settings::ReadConfigValue("CheckForUpdates", checkUpdates))
```

Comment:
In this place, where you lost the `else` branch (which wouldn’t make sense here), it doesn’t make sense to check if the value could be read with the double-if and `Settings::ReadConfigValue("CheckForUpdates”, checkUpdates, false);`, i.e. using a default value, and ignoring the return value makes more sense.